### PR TITLE
Workflow Consistency / Simplification Updates in All Workflows - Part1

### PR DESCRIPTION
**Pull request by peterzhuamazon**
This PR is to improve workflow consistency / simplification updates in all workflows.
* Unifying naming conventions of workflow files here
* Unifying workflow job names with RPM one for references
* Simplify workflow integTests by utilizing the setup_runner_service.sh scripts
* Add a setup_runner_service_windows powershell script for windows env
(windows GitHub runner default shell is set to powershell: https://github.blog/changelog/2019-10-17-github-actions-default-shell-on-windows-runners-is-changing-to-powershell/)

### Test Cases: ###
* RPM: https://github.com/opendistro-for-elasticsearch/opendistro-build/actions/runs/187850756
* WINDOWS: https://github.com/opendistro-for-elasticsearch/opendistro-build/actions/runs/187880741

This PR consumes the changes from @weicongs-amazon 's windows ad kibana plugin integration test: https://github.com/opendistro-for-elasticsearch/opendistro-build/pull/332

Another PR for Part2 of other distros will be created later.

Thanks.

### CODE REVIEW

File: .github/workflows/test-build-deb.yml [hunk 1]
```diff
@@ -593,7 +593,7 @@ jobs:
         with:
           repository: opendistro-for-elasticsearch/kibana-oss
           ref: ${{env.es_version}}
-          token: ${{ secrets.ODFE_RELEASE_BOT_PRIVATE_READ_WRITE_TOKEN }}
+          token: ${{ secrets.ODFE_RELEASE_BOT_PUBLIC_PRIVATE_READ_WRITE_TOKEN }}
```
Comment (gaiksaya) [inline .github/workflows/test-build-deb.yml:596]: Why did we change the token again?

**Comment by gaiksaya** [comment]
Also can you add the introduction at the start to tell what the script does?  for both runners script. We should do that for all of them

File: .github/workflows/test-build-deb.yml (hunk 1 above)
Comment (peterzhuamazon) [reply .github/workflows/test-build-deb.yml:596]: @gaiksaya Apparently after testing the private only token does not work at all, and is no need to create a separate token just for a reading purpose. I just use the same token for all now. If we have access to private there is no reason it cannot be used for public. For public read only there is another token for that purpose.

**Comment by peterzhuamazon** [comment]
> Also can you add the introduction at the start to tell what the script does? for both runners script. We should do that for all of them

@gaiksaya Will add another PR later for scripts documentation.

**Pull request merged**